src/Components/App/App.tsx

```diff
@@ -73,22 +73,17 @@ class App extends React.Component<{}, State> {
                   />
                 )
               } else {
+                const header = this.state.favorites.length ? "Favorites!" : "You have no favorites"
                 return (
-                  this.state.favorites.length ?
                   <Bookshelf
-                    favoritesHeader='Favorites!'
-                    favoriteBooks={this.state.favorites}
-                  />
-                  : !this.state.favorites.length &&
-                  <Bookshelf
-                    favoritesHeader='You have no favorites!'
+                    favoritesHeader={header}
                     favoriteBooks={this.state.favorites}
                   />
                 )
               }
             }}
           />
-          
+
         </Switch>
       </main>
     )
```

Comment:
Awesome job handling these merge conflicts and getting everything to work. Also, I love you're refactor of my conditional logic by putting it all in a variable called header.